.github/workflows/docker-publish.yml

```
@@ -0,0 +1,35 @@
+name: Docker Publish
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+# From https://docs.github.com/en/actions/publishing-packages/publishing-docker-images
+jobs:
+  docker-publish:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+    steps:
+    - uses: actions/checkout@v3
+    - name: Log in to the Container registry
+      uses: docker/login-action@v3
+      with:
+        registry: ghcr.io
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+    - name: Extract metadata (tags, labels) for Docker
+      id: meta
+      uses: docker/metadata-action@v5
+      with:
+        images: ghcr.io/${{ github.repository }}
+    - name: Build and push(if main and not PR) Docker image
```

Minor: please add spaces around the parentheses

```suggestion
    - name: Build and push ( if on main ) Docker image
```